File: 프로그래머스/1/42840. 모의고사/모의고사.py
def solution(answers):
    student1 = [1,2,3,4,5]
    student2 = [2,1,2,3,2,4,2,5]
    student3 = [3,3,1,1,2,2,4,4,5,5]
    
    s1, s2, s3 = 0,0,0
    for i in range(len(answers)):
        if answers[i] == student1[i%5]:
            s1 += 1
        if answers[i] == student2[i%8]:
            s2 += 1
        if answers[i] == student3[i%10]:
            s3 += 1
    
    answers = []
    max_value = max(s1,s2,s3)
    if s1 == max_value:
        answers.append(1)
    if s2 == max_value:
        answers.append(2)
    if s3 == max_value:
        answers.append(3)
        
    return answers

File: 프로그래머스/1/42840. 모의고사/모의고사.swift
import Foundation

func solution(_ answers:[Int]) -> [Int] {
    let student1 = [1,2,3,4,5]
    let student2 = [2, 1, 2, 3, 2, 4, 2, 5]
    let student3 = [3, 3, 1, 1, 2, 2, 4, 4, 5, 5]
    
    var ans1 = 0
    var ans2 = 0
    var ans3 = 0
    
    for i in 0..<answers.count {
        if answers[i]==student1[i%5] {
            ans1 += 1
        }
        if answers[i]==student2[i%8] {
            ans2 += 1
        }
        if answers[i]==student3[i%10] {
            ans3 += 1
        }
    }
    
    var res: [Int] = []
    let maxNum = max(ans1, ans2, ans3)
    if maxNum == ans1 {
        res.append(1)
    }
    if maxNum == ans2 {
        res.append(2)
    }
    if maxNum == ans3 {
        res.append(3)
    }
    return res
}